### ISSUE TITLE
RGB input instead of grayscale

### ISSUE THREAD

**Issue by zachpvin** [lRomul/ball-action-spotting#1]
I wonder do you have any tip on tweaking this for RGB input, especially the neighboring stacking part.
https://github.com/lRomul/ball-action-spotting/blob/73ae389db6d1b7dd32a94f69e1942c96ce419fa4/src/models/multidim_stacker.py#L214
I can't figure out how to make this works. My idea was to proceed with concatenating on the channel still, and it would result in 
`(b*num_stacks, stack_size*rgb_channel, h, w)` = `(2*5, 3*3, 736, 1280)` before feeding into the 2D encoder. Though dimension-wise this works, but theory-wise I'm having doubt.

Appreciate any comment on this.

**Comment by lRomul** [comment]
Hi! I implemented your idea, and it works. I think such a model can be trained well.

```python
import torch

from src.models.multichan_stacker import MultiChanStacker

num_frames = 15
num_chans = 3

nn_module = MultiChanStacker(
    model_name="tf_efficientnetv2_b0", 
    num_classes=2, 
    num_frames=num_frames, 
    num_chans=num_chans
)
input_tensor = torch.zeros(5, num_frames, num_chans, 224, 224)
output = nn_module(input_tensor)
```

**Comment by lRomul** [comment]
I have not experimented with the model besides using it for transfer learning. But I have done several experiments without changing architecture parameters too much to get better initial weights. You can find experiments [here](https://docs.google.com/spreadsheets/d/1mGnTdrVnhoQ8PJKNN539ZzhZxSowc4GpN9NdyDJlqYo/edit#gid=2136737334). My in-training average precision metric was high, but tight average-mAP from the challenge was low. I haven't figured out what the problem is. There may be a bug somewhere in the predictions or evaluation. Or for a successful result, the model and post-processing parameters must be changed significantly.

I agree with longer sequences. I think my version is more focused on short-term actions. You can take a look at the solution to the problem with long sequences using a similar architecture [here](https://www.kaggle.com/competitions/nfl-player-contact-detection/discussion/391740).
The weak linear classifier may be acceptable here because features from different time stacks start fusing in the 3D encoder. We can add more 3D layers or even blocks to increase receptive field and generalization. But it might be worth replacing the linear classifier with a recurrent layer. I think that it very much depends on the task and the dataset.
